crdt.go

```diff
@@ -0,0 +1,532 @@
+// Package crdt provides a replicated go-datastore implementation using
+// Merkle-CRDTs built with IPLD nodes. This Datastore is agnostic to how new
+// MerkleDAG roots are broadcasted to the rest of replicas (Broadcaster
+// component) and to how the IPLD nodes are made discoverable and retrievable
+// to by other replicas (DAG-Sync component).
+package crdt
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	pb "github.com/ipfs/go-ds-crdt/pb"
+
+	blockfmt "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	query "github.com/ipfs/go-datastore/query"
+	dshelp "github.com/ipfs/go-ipfs-ds-help"
+	logging "github.com/ipfs/go-log"
+	"github.com/pkg/errors"
+)
+
+var _ ds.Datastore = (*Datastore)(nil)
+var _ ds.Batching = (*Datastore)(nil)
+
+// datastore namespace keys. Short keys save space and memory.
+const (
+	blocksNs = "b" // blocks
+	headsNs  = "h" // heads
+	setNs    = "s" // set
+)
+
+var (
+	ErrNoMoreBroadcast = errors.New("Receiving blocks aborted since no new blocks will be broadcasted.")
+)
+
+type Broadcaster interface {
+	Broadcast([]byte) error
+	Next() ([]byte, error)
```

Comment:
```suggestion
	Next() (context.Context, []byte, error)
```
Next should be responsible for the deserializing of the context from the payload, if one exists.